docs: add FT232R/FT245R, FT232H, FT-X. (#6)
wanda-phi/prjspeedbump: docs/src/ft232h.md

| ADBUS6 | DCD#    | D6      | D6        | D6       | D6      | GPIOL2  | -             | CIOPIO6 |
| ADBUS7 | RI#     | D7      | D7        | D7       | D7      | GPIOL3  | -             | CIOPIO7 |
| ACBUS0 | [CBUS0] | RXF#    | RXF#      | CS#      | [CBUS0] | GPIOH0  | [CBUS0]       | SCLK    |
| ACBUS1 | [CBUS1] | TXE#    | TXE#      | A0       | WR#     | GPIOH1  | [CBUS1]       | CS#     |
| ACBUS2 | [CBUS2] | RD#     | RD#       | RD#      | RD#     | GPIOH2  | [CBUS2]       | CIPO    |
| ACBUS3 | [CBUS3] | WR      | WR        | WR#      | [CBUS3] | GPIOH3  | [CBUS3]       | [CBUS3] |
| ACBUS4 | [CBUS4] | SIWU#   | SIWU#     | SIWU#    | SIWU#   | GPIOH4  | SIWU#         | [CBUS4] |
| ACBUS5 | [CBUS5] | [CBUS5] | CLKOUT    | [CBUS5]  | [CBUS5] | GPIOH5  | [CBUS5]       | [CBUS5] |
| ACBUS6 | [CBUS6] | [CBUS6] | OE#       | [CBUS6]  | [CBUS6] | GPIOH6  | [CBUS6]       | [CBUS6] |
| ACBUS7 | PWRSAV# | PWRSAV# | PWRSAV#   | PWRSAV#  | PWRSAV# | GPIOH7  | PWRSAV#       | PWRSAV# |
| ACBUS8 | [CBUS8] | [CBUS8] | [CBUS8]   | [CBUS8]  | [CBUS8] | [CBUS8] | [CBUS8]       | [CBUS8] |
| ACBUS9 | [CBUS9] | [CBUS9] | [CBUS9]   | [CBUS9]  | [CBUS9] | [CBUS9] | [CBUS9]       | [CBUS9] |

> 

Reviewer: ```suggestion
Note: the function of the pins labeled with square brackets is determined by the configuration in EEPROM when the relevant mode is active.
```

| `0x5`        | all but CBUS7 | SLEEP#                      |
| `0x6`        | all but CBUS7 | const-0 output              |
| `0x7`        | CBUS[05689]   | const-1 output              |
| `0x8`        | CBUS[5689]    | I/O (for CBUS bitbang mode) |
| `0x9`        | all but CBUS7 | UART TXDEN                  |
| `0xa`        | CBUS[05689]   | CLK30                       |
| `0xb`        | CBUS[05689]   | CLK15                       |
| `0xc`        | CBUS[05689]   | CLK7.5                      |

TODO: valid pin set far from certain.

Note that the EEPROM-configured CBUS pin function may be overriden by whatever mode is currently active.

## EEPROM data format

The device will accept 128-word and 256-word EEPROMs.  In case of 256-word EEPROMs, only first half is covered by the checksum.  The other half can still be used to store user data.

For USB-PD devices, a 256-word EEPROM is required for the USB-PD part of the device to function.  The second half of the EEPROM is used to store the USB-PD configuration data, and it has its own, separate checksum.  It is described on the [USB-PD page](usb-pd.md#eeprom-data-format).

> Note that 64-word EEPROMs are not supported for unknown reasons.  This may be related to the funny fixed word at address `0x45`.  Since this is in the same address range as the FT232R uses for factory-programmed data, it is somewhat likely that it is an accidentally merged bit of FT232R circuitry.

Reviewer: ```suggestion
Note that 64-word EEPROMs are not supported for unknown reasons.  This may be related to the funny fixed word at address `0x45`.  Since this is in the same address range as the FT232R uses for factory-programmed data, it is somewhat likely that it is an accidental bit of leftover FT232R circuitry.
```

The EEPROM format is:

- word 0x00:
  - bits 0-3: mode
    - 0: UART
    - 1: 245-style FIFO
    - 2: CPUFIFO
    - 4: opto-isolated
    - 8: FT1248
  - bit 4: enable bind to VCP driver
  - bit 8: FT1248 clock polarity (CPOL)
    - 0: low

>     - 1: high

Reviewer: ```suggestion
    - 1: high
    - TODO: check what this means exactly in hardware
```

>   - bit 9: FT1248 bit order
>     - 0: MSB
>     - 1: LSB

Reviewer: ```suggestion
  - bit 9: FT1248 bit order
    - 0: MSB first
    - 1: LSB first
```

>   - bit 10: FT1248 flow control disabled

Reviewer: To be consistent:

```suggestion
  - bit 10: disable FT1248 flow control
```
